addons/purchase/models/purchase.py

```diff
@@ -525,6 +533,14 @@ def _compute_amount(self):
                 'price_subtotal': taxes['total_excluded'],
             })
 
+    @api.multi
+    def _compute_tax_id(self):
+        for line in self:
+            fpos = line.order_id.fiscal_position_id or line.order_id.partner_id.property_account_position_id
```

Comment:
This has not been filtered with `force_company` argument, so in case this is done with admin, the fiscal position brought will be incorrect. For fixing it:
```
fpos = line.order_id.fiscal_position_id
if not fpos:
    order = line.order_id
    partner = order.partner_id.with_context(force_company=order.company_id.id)
    fpos = partner.property_account_position_id
```